README.md

```diff
@@ -1,83 +1,250 @@
-emoji.hasEmoji('🍕') // Validate if this library knows an emoji like `🍕`
-emoji.hasEmoji('pizza') // Validate if this library knowns a emoji with the name `pizza`
-emoji.strip('⚠️ 〰️ 〰️ low disk space') // Strips the string from emoji's, in this case returns: "low disk space".
-emoji.replace('⚠️ 〰️ 〰️ low disk space', (emoji) => `${emoji.key}:`) // Replace emoji's by callback method: "warning: low disk space"
+emoji.get(':unicorn:')
+//=> 🦄
+
+emoji.which('🦄')
+//=> "unicorn"
+
+emoji.emojify('I :heart: :coffee:!')
+//=> "I ❤️ ☕️!"
+
+emoji.random()
+//=> { name: 'house', emoji: '🏠' }
+```
+
+## API
+
+### emoji.get(name)
+
+Get an emoji from an emoji name.
+
+#### name
+
+Type: `string`
+
+The name of the emoji to get.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.get('unicorn'))
+//=> '🦄'
+```
+
+### emoji.which(emoji)
+
+Get an emoji name from an emoji.
+
+#### emoji
+
+Type: `string`
+
+The emoji to get the name of.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.which('🦄'))
+//=> 'unicorn'
+```
+
+### emoji.random()
+
+Get a random emoji.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.random())
+//=> { name: 'unicorn', emoji: '🦄' }
+```
+
+### emoji.search(keyword)
+
+Search for emojis containing the provided name in their name.
+
+#### keyword
+
+Type: `string`
+
+The keyword to search for.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.search('honey'))
+//=> [ { name: 'honeybee', emoji: '🐝' }, { name: 'honey_pot', emoji: '🍯' } ]
+```
+
+### emoji.find(emoji)
+
+Get the name and character of an emoji.
+
+#### emoji
+
+Type: `string`
+
+The emoji to get the data of.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.find('🦄'))
+//=> { name: 'unicorn', emoji: '🦄' }
 ```
 
-## Options
+### emoji.has(emoji)
 
-### onMissing
-`emoji.emojify(str, onMissing)`
+Check if this library supports a specific emoji.
 
-As second argument, `emojify` takes an handler to parse unknown emojis. Provide a function to add your own handler:
+#### emoji
+
+Type: `string`
+
+The emoji to check.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.has('🦄'))
+//=> true
+```
+
+### emoji.replace(input, replacement)
+
+Replace the emojis in a string.
+
+#### input
+
+Type: `string`
+
+The input string.
+
+#### replacement
+
+Type: `string | (emoji: string, index: number, string: InputValueType) => string`
+
+The character to replace the emoji with. Can be either a string or a callback that returns a string.
+
+```js
+const emoji = require('node-emoji')
+
+console.log(emoji.replace('The 🦄 is a fictitious animal.', 'unicorn'))
+//=> 'The unicorn is a fictitious animal.'
+```
+
+### emoji.strip(input, options?)
+
+Remove all of the emojis from a string.
+
+#### input
+
+Type: `string`
+
+The input string to strip the emojis from.
+
+#### options
+
+Type: `object`
+
+##### removeSpaces
+
+Type: `boolean`\
+Default: `true`.
```

Comment:
I've always found it confusing to have APIs that default a boolean parameter to `true` when not provided. Can we call this:

```suggestion
##### preserveSpaces

Type: `boolean`\
Default: `false`.
```